Merge advanced security workshops into  docs/codeql/codeql-workshops/ (#7379)
github/codeql: docs/codeql/codeql-workshops/common/cli-for-codeql.org

* Introduction

>   As for any compiler / libary pair, only certain versions of the codeql compiler

Reviewer: ```suggestion
  As for any compiler and library pair, only certain versions of the CodeQL compiler
```

>   will work with certain versions of the libary.  To ensure compatibility, the

Reviewer: ```suggestion
  will work with certain versions of the library.  To ensure compatibility, the
```

>   tools creating the db, the query writers (if any), and the query runners should

Reviewer: ```suggestion
  tools creating the database, the query writers (if any), and the query runners should
```
